apps/system/js/sync_manager.js

```diff
@@ -0,0 +1,547 @@
+    'iac-' + SYNC_MANAGEMENT_API_IAC_KEYWORD,
+
+    /** Sync State Machine events **/
+    'onsyncdisabled',
+    'onsyncenabled',
+    'onsyncenabling',
+    'onsyncerrored',
+    'onsyncsyncing',
+
+    /** Sync app lifecycle **/
+    'killapp',
+  ];
+
+  SyncManager.SUB_MODULES = [
+    'SyncStateMachine'
+  ];
+
+  BaseModule.create(SyncManager, {
+    name: 'SyncManager',
+    DEBUG: true,
+    EVENT_PREFIX: 'sync',
+    settings: new Map(),
+    store: new Map(),
+
+    /** BaseModule life cycle **/
+
+    _start: function() {
+      return new Promise((resolve, reject) => {
+        LazyLoader.load('/shared/js/sync/errors.js', () => {
+          navigator.mozSetMessageHandler('request-sync', event => {
+            this.debug('Sync request');
+            if (event.task != SYNC_TASK) {
+              return;
+            }
+            Service.request('SyncStateMachine:sync');
+          });
+
+          SyncManager.SETTINGS.forEach(setting => {
+            this['_observe_' + setting] = value => {
+              this.debug('Observed ' + setting + ' changed to ' + value);
+              this.settings.set(setting, value);
+            };
+          });
+
+          this.initStore().then(() => {
+            this.debug('Sync.state setting initial value', this.state);
+            // SyncStateMachine starts by default with 'disabled', so we can
+            // request to enable sync or stay in the disabled state.
+            // We need to make sure that we didn't end up in an inconsistent
+            // state during the latest run (i.e. error, success, syncing,
+            // enabling).
+            switch(this.state) {
+              case 'enabled':
+              case 'success':
+              case 'syncing':
+                Service.request('SyncStateMachine:enable').then(resolve);
+                break;
+              case 'enabling':
+              case 'errored':
+                this.saveState('disabled');
+                resolve();
+                break;
+              case 'disabled':
+                resolve();
+                break;
+              default:
+                console.error('Unexpected sync.state value', this.state);
+                reject();
+                break;
+            }
+          });
+        });
+      });
+    },
+
+    _stop: function() {},
+
+    /** Event handlers **/
+
+    // We expose a very simple IAC based API to allow the Settings app to
+    // manage Sync.
+    //
+    // We expect IAC messages to contain a property named 'name' with any
+    // of the following values corresponding to the requests that we are
+    // able to handle:
+    //   - enable
+    //   - disable
+    //   - sync
+    //   - getInfo
+    //
+    // SyncStateMachine is responsible of checking if the request is valid in
+    // the current situation and progressing it accordingly by publishing the
+    // corresponding events.
+    '_handle_iac-gaia::sync::management': function(event) {
```

Comment:
Can we use a simpler channel name? This method name is banana :)
I suggest `sync-management` but if you really want the namespacing `gaia-sync-management` would at least prevent the introduction of a _third_ special character in the method name.